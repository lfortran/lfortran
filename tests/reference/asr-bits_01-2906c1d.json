{
    "basename": "asr-bits_01-2906c1d",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/bits_01.f90",
    "infile_hash": "4fdd6c630a9f2559417372a42cca1dc1c98fa7af4e4a5442ebf9ca96",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-bits_01-2906c1d.stdout",
    "stdout_hash": "e4d6face07cc10b32e921ee0fe9224fa945d2a66afc29f86f0e375c9",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}