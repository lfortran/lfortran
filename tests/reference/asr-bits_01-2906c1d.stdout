(TranslationUnit
    (SymbolTable
        1
        {
            bits_01:
                (Program
                    (SymbolTable
                        2
                        {
                            ibclr:
                                (ExternalSymbol
                                    2
                                    ibclr
                                    6 ibclr
                                    lfortran_intrinsic_bit
                                    []
                                    ibclr
                                    Private
                                ),
                            ibclr@ibclr32:
                                (ExternalSymbol
                                    2
                                    ibclr@ibclr32
                                    6 ibclr32
                                    lfortran_intrinsic_bit
                                    []
                                    ibclr32
                                    Private
                                ),
                            ibclr@ibclr64:
                                (ExternalSymbol
                                    2
                                    ibclr@ibclr64
                                    6 ibclr64
                                    lfortran_intrinsic_bit
                                    []
                                    ibclr64
                                    Private
                                ),
                            ibset:
                                (ExternalSymbol
                                    2
                                    ibset
                                    6 ibset
                                    lfortran_intrinsic_bit
                                    []
                                    ibset
                                    Private
                                ),
                            ibset@ibset32:
                                (ExternalSymbol
                                    2
                                    ibset@ibset32
                                    6 ibset32
                                    lfortran_intrinsic_bit
                                    []
                                    ibset32
                                    Private
                                ),
                            ibset@ibset64:
                                (ExternalSymbol
                                    2
                                    ibset@ibset64
                                    6 ibset64
                                    lfortran_intrinsic_bit
                                    []
                                    ibset64
                                    Private
                                ),
                            ieor:
                                (ExternalSymbol
                                    2
                                    ieor
                                    6 ieor
                                    lfortran_intrinsic_bit
                                    []
                                    ieor
                                    Private
                                ),
                            ieor@ieor32:
                                (ExternalSymbol
                                    2
                                    ieor@ieor32
                                    6 ieor32
                                    lfortran_intrinsic_bit
                                    []
                                    ieor32
                                    Private
                                ),
                            ieor@ieor64:
                                (ExternalSymbol
                                    2
                                    ieor@ieor64
                                    6 ieor64
                                    lfortran_intrinsic_bit
                                    []
                                    ieor64
                                    Private
                                ),
                            int32:
                                (ExternalSymbol
                                    2
                                    int32
                                    4 int32
                                    lfortran_intrinsic_iso_fortran_env
                                    []
                                    int32
                                    Public
                                ),
                            int64:
                                (ExternalSymbol
                                    2
                                    int64
                                    4 int64
                                    lfortran_intrinsic_iso_fortran_env
                                    []
                                    int64
                                    Public
                                )
                        })
                    bits_01
                    [iso_fortran_env]
                    [(If
                        (IntegerCompare
                            (FunctionCall
                                2 ibclr@ibclr32
                                2 ibclr
                                [((IntegerConstant 3 (Integer 4 [])))
                                ((IntegerConstant 0 (Integer 4 [])))]
                                (Integer 4 [])
                                (IntegerConstant 2 (Integer 4 []))
                                ()
                            )
                            NotEq
                            (IntegerConstant 2 (Integer 4 []))
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (IntegerCompare
                            (FunctionCall
                                2 ibclr@ibclr64
                                2 ibclr
                                [((IntegerConstant 4 (Integer 8 [])))
                                ((IntegerConstant 1 (Integer 4 [])))]
                                (Integer 8 [])
                                (IntegerConstant 4 (Integer 8 []))
                                ()
                            )
                            NotEq
                            (IntegerConstant 4 (Integer 8 []))
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (IntegerCompare
                            (FunctionCall
                                2 ibset@ibset32
                                2 ibset
                                [((IntegerConstant 1 (Integer 4 [])))
                                ((IntegerConstant 2 (Integer 4 [])))]
                                (Integer 4 [])
                                (IntegerConstant 5 (Integer 4 []))
                                ()
                            )
                            NotEq
                            (IntegerConstant 5 (Integer 4 []))
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (IntegerCompare
                            (FunctionCall
                                2 ibset@ibset64
                                2 ibset
                                [((IntegerConstant 2 (Integer 8 [])))
                                ((IntegerConstant 3 (Integer 4 [])))]
                                (Integer 8 [])
                                (IntegerConstant 10 (Integer 8 []))
                                ()
                            )
                            NotEq
                            (IntegerConstant 10 (Integer 8 []))
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (IntegerCompare
                            (FunctionCall
                                2 ieor@ieor32
                                2 ieor
                                [((IntegerConstant 16 (Integer 4 [])))
                                ((IntegerConstant 2 (Integer 4 [])))]
                                (Integer 4 [])
                                (IntegerConstant 18 (Integer 4 []))
                                ()
                            )
                            NotEq
                            (IntegerConstant 18 (Integer 4 []))
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (IntegerCompare
                            (FunctionCall
                                2 ieor@ieor64
                                2 ieor
                                [((IntegerConstant 31 (Integer 8 [])))
                                ((IntegerConstant 3 (Integer 8 [])))]
                                (Integer 8 [])
                                (IntegerConstant 28 (Integer 8 []))
                                ()
                            )
                            NotEq
                            (IntegerConstant 28 (Integer 8 []))
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                ),
            iso_fortran_env:
                (IntrinsicModule lfortran_intrinsic_iso_fortran_env),
            lfortran_intrinsic_bit:
                (IntrinsicModule lfortran_intrinsic_bit)
        })
    []
)
